src/workerd/jsg/jsg.c++

```diff
@@ -149,6 +150,13 @@ void Lock::setLoggerCallback(kj::Function<Logger>&& logger) {
   IsolateBase::from(v8Isolate).setLoggerCallback({}, kj::mv(logger));
 }
 
+void Lock::requestGcForTesting() const {
+  if (!isPredictableModeForTest()){
+    kj::throwFatalException(KJ_EXCEPTION(UNIMPLEMENTED, "Test GC used while not in a test"));
```

Comment:
nit; Maybe `UNIMPLEMENTED` is a bit misleading here.

Reply:
The other option here would be `FAILED`, but seems to send not the right signal as well. Optimally there would be a `WILLNOTIMPLEMENT` type. Do you think `FAILED` would be better?

Reply:
This should be `KJ_FAIL_REQUIRE()`, this is exactly what `REQUIRE` is for -- enforcing the interface contract.

Reply:
Or even just:

```
KJ_REQUIRE(isPredictableModeForTest(), "test GC used while not in a test");
```

Reply:
That does work out better. I was going for making it super obvious that someone made a bad decision, but `KJ_REQUIRE` does give debug info as well so I'll change it over.

Reply:
Done